### PR TITLE
Minimal Fetch API polyfill

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #657

/cc @erwinmombay

### CODE REVIEW

File: src/xhr.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+      // TODO(dvoytenko): This is currently simplified: we will wait for the
+      // whole document loading to complete. This is fine for the use cases
+      // we have now, but may need to be reimplemented later.
+      if (xhr.readyState == /* COMPLETE */ 4) {
+        resolve(new FetchResponse(xhr));
+      }
+    };
+    xhr.onerror = () => {
+      reject(new Error('Network failure'));
```
Comment (cramforce) [inline src/xhr.js]: Forward error passed into onerror?

Comment (dvoytenko) [inline src/xhr.js]: onerror's argument is an event. From what I tried to find - I don't see any errors there. Just basic event info with `type=error` and no real additional information.

File: src/xhr.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+ * @return {!Promise<!FetchResponse>}
+ */
+function fetch(input, opt_init) {
+  assert(typeof input == 'string', 'Only URL supported: %s', input);
+  const init = opt_init || {};
+  assert(!init.credentials || init.credentials == 'include',
+      'Only credentials=include support: %s', init.credentials);
+
+  return new Promise(function(resolve, reject) {
+    const xhr = createXhrRequest(init.method || 'GET', input);
```
Comment (cramforce) [inline src/xhr.js]: should this use fetch where available?

Comment (dvoytenko) [inline src/xhr.js]: Hmm. Good point. We can certainly use fetch in this case. I'll prioritize `window.fetch` when available.

Comment (dvoytenko) [inline src/xhr.js]: Done. PTAL.

File: src/xhr.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+ * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+ *
+ * Notice that the "fetch" method itself is not exported as that would require
+ * us to immediately support a much wide API.
+ *
+ * @param {string} input
+ * @param {?FetchInit=} opt_init
+ * @return {!Promise<!FetchResponse>}
+ */
+const fetch = window.fetch || function fetch(input, opt_init) {
```
Comment (cramforce) [inline src/xhr.js]: Since we have no coverage of testing in non-Chrome, please make a mode here where the tests run with the polyfill.

Comment (dvoytenko) [inline src/xhr.js]: I switched to Xhr service and provided both sets of tests: native and polyfill.

**Comment by cramforce** [comment]
LGTM

**Pull request merged**